ascend-sdk/src/main/java/ai/evolv/ascend/android/Allocator.java

```diff
@@ -36,19 +72,112 @@ HttpUrl createAllocationsUrl() {
                 .build();
     }
 
-    synchronized JsonArray fetchAllocations() throws AscendAllocationException {
-        JsonArray allocations;
+    Future<JsonArray> fetchAllocations() {
+        OkHttpClient client = new OkHttpClient.Builder()
+                .callTimeout(config.getTimeout(), TimeUnit.MILLISECONDS)
+                .retryOnConnectionFailure(false)
+                .build();
+
+        final Request request = new Request.Builder()
+                .url(createAllocationsUrl())
+                .build();
+
+        final SettableFuture<JsonArray> responseFuture = SettableFuture.create();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override public void onFailure(Call call, IOException e) {
+                resolveAllocationFailure(responseFuture);
+            }
+
+            @Override public void onResponse(Call call, Response response) {
+                String body = "";
+                JsonArray allocations = new JsonArray();
+                try {
+                    try {
```

Comment:
You're just catching the exception thrown in the outer try. You should move this try to the same level as the outer try.

Reply:
The point of the nested try is to skip the logic outside of the inner try in the event the the inner try fails. I feel like this is the better way to go.